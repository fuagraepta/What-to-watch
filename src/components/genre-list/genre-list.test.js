import React from 'react';
import renderer from 'react-test-renderer';
import GenreList from './genre-list.jsx';

const mockGenres = [`All genres`, `Crime`, `Horror`];

describe(`The component is rendered correctly`, () => {
  it(`Genre list correctly renders with transferred mock genres`, () => {
    const genreList = renderer
    .create(<GenreList
      allGenres = {mockGenres}
      onGenreClick = {jest.fn()}
      onItemActivate = {jest.fn()}
      activeItem = {`All genres`}
    />)
      .toJSON();

    expect(genreList).toMatchSnapshot();
  });
});
