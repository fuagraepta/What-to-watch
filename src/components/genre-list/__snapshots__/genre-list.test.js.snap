// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Genre list correctly renders with transferred mock genres 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
</ul>
`;
